### PR TITLE
add day 7

### PR DESCRIPTION
**Pull request by wtchangdm**

### CODE REVIEW

File: 2023/07/wtchangdm/day7.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        if result != Ordering::Equal {
+            return result;
+        }
+
+        for (self_label, other_label) in self.labels.iter().zip(other.labels.iter()) {
+            let label_cmp = self_label.cmp(other_label);
+            if label_cmp != Ordering::Equal {
+                return label_cmp;
+            }
+        }
```
Comment (weihanglo) [inline 2023/07/wtchangdm/day7.rs:148]: here is [a default `Ord` impl for `Vec`](https://doc.rust-lang.org/1.74.1/std/vec/struct.Vec.html#impl-Ord-for-Vec%3CT,+A%3E), so should be sufficient to this case?

Something like

```
    fn cmp(…) -> Ordering {
        (self.hand_type.rank(), self.labels).cmp(&(other.hand_type.rank(), other.labels))
    }
```

**Review by weihanglo — COMMENTED** [review]
Not looking into the correctness. Just some suggestion for (perhaps) more idiomatic Rust.

**Comment by weihanglo** [comment]
👍🏾 

Merge this and feel free to improve your solution :)

**Review by weihanglo — APPROVED** [review]

**Comment by wtchangdm** [comment]
@weihanglo Thanks for the advice again! I am definitely new to cmp here.
Will revise later.

File: 2023/07/wtchangdm/day7.rs (hunk 1 above)
Comment (wtchangdm) [reply 2023/07/wtchangdm/day7.rs:148]: This is actually cool. Never thought it can work this way (also not sure where to find good examples).

**Pull request merged**